paywall/src/data-iframe/cacheHandler.js

```diff
@@ -4,35 +4,89 @@ import linkKeysToLocks from './blockchainHandler/linkKeysToLocks'
 let currentNetwork
 let currentAccount
 
+const nullAccount = '0x0000000000000000000000000000000000000000'
+
 export const getAccount = cache.getAccount
 export const getNetwork = cache.getNetwork
 
+export async function setAccount(window, account) {
+  // intercept the account setting so we have it available as well for retrieving user-specific cache
+  currentAccount = account
+  return cache.setAccount(window, account)
+}
+
+export async function setNetwork(window, network) {
+  // intercept the network setting so we have it available as well for retrieving user-specific cache
+  currentNetwork = network
+  return cache.setNetwork(window, network)
+}
+
+export async function getAccountBalance(window) {
+  return _get(window, 'balance')
+}
+
+export async function setAccountBalance(window, balance) {
+  return _put(window, 'balance', balance)
+}
+
 export function setup(networkId, account) {
   currentNetwork = networkId
   currentAccount = account
 }
 
-async function _get(window, key) {
```

Comment:
extend the definition of `_get` and `_put` to pass the null account for storing things like locks that are the same across different accounts. This allows centralizing the important stuff here at the top of the file, and the `getLocks` `setKeys` etc. can focus on the logic of what they do instead of low-level cache management